src/mapper.c

```diff
@@ -0,0 +1,209 @@
+#include <stdio.h>
+#include <linux/input.h>
+#include <linux/uinput.h>
+
+#include "queue.h"
+#include "keys.h"
+#include "config.h"
+#include "emit.h"
+#include "mapper.h"
+
+// The state machine state
+enum states state = idle;
+
+// Flag if the hyper key has been emitted
+static int hyperEmitted;
+
+/**
+ * Checks if the key is the hyper key.
+ * */
+static int isHyper(int code)
+{
+    return code == hyperKey;
+}
+
+/**
+ * Checks if the key has been mapped.
+ * */
+static int isMapped(int code)
+{
+    return keymap[code] != 0;
+}
+
+/**
+ * Converts the input code to the mapped code.
+ * */
+static int getMapped(int code)
+{
+    return keymap[code];
+}
+
+/**
+ * Checks if the key has been permanently mapped.
+ * */
+static int isRemapped(int code)
+{
+    return remap[code] != 0;
+}
+
+/**
+ * Converts the input code to the remapped code.
+ * */
+static int getRemapped(int code)
+{
+    if (remap[code] != 0)
+    {
+        return remap[code];
+    }
+    return code;
+}
+
+/**
+ * * Processes a key input event. Converts and emits events as necessary.
+ * */
+void processKey(int type, int code, int value)
+{
+    printf("processKey(in): code=%i value=%i state=%i\n", code, value, state);
```

Comment:
I suppose the `printf` calls in this function should not be in the final production code, right? However, I would keep them here commented out for when I manage to implement debugging options (as in #42). Then, they can be hidden behind `debug active` constraint.

Reply:
I commented them out again for now.